packages/web3-core-method/src/index.js

```diff
@@ -314,13 +314,35 @@ Method.prototype._confirmTransaction = function (defer, result, payload, extraFo
             utils._fireError({message: 'Failed to subscribe to new newBlockHeaders to confirm the transactions receipt. Are you using HttpProvider? Please switch to Websockets.', data: err}, defer.eventEmitter, defer.reject);
         }
     };
-
+    
+  // start watching for confirmation depending on the support features of the provider
+  var startWatching = function() {
     // if provider allows PUB/SUB
     if (_.isFunction(this.requestManager.provider.on)) {
-        method.eth.subscribe('newBlockHeaders', checkConfirmation);
+      method.eth.subscribe('newBlockHeaders', checkConfirmation);
     } else {
-        intervalId = setInterval(checkConfirmation, 1000);
+      intervalId = setInterval(checkConfirmation, 1000);
     }
+  };
+
+  // first check if we already have a confirmed transaction
+  method.eth.getTransactionReceipt(result)
+    .then(function(receipt) {
+      if (receipt && receipt.blockNumber) {
+        checkConfirmation(null, receipt.blockNumber, null, receipt);
```

Comment:
but since the confirm-function is used as callback in the subscribe-function ( ``` method.eth.subscribe('newBlockHeaders', checkConfirmation); ``` 
the 2nd argument will be the blockheader, so we can't remove the block-parameter if we want to access the sub-argument. But I am passing a null now.